src/nl_to_polynomial.jl

```diff
@@ -94,7 +94,11 @@ function _to_polynomial!(
         return a^b
     elseif _is_operator(expr, :/) && length(operands) == 2
         a, b = _to_polynomial!.(Ref(d), T, operands)
-        return a / b
+        if iszero(MP.rem(a, b))
```

Comment:
You can call `Base.divrem` otherwise you computing it twice since both `div` and `rem` end up calling `divrem`